### ISSUE TITLE
NEST v2.12.0 does not build on x86

### ISSUE THREAD

**Issue by ikitayama** [nest/nest-simulator#934]

**Comment by ikitayama** [comment]
```-bash-4.2$ make
Scanning dependencies of target generate_help
Extracting help information; this may take a litte while.
Traceback (most recent call last):
  File "generate_help.py", line 102, in <module>
    require = check_ifdef(item, filetext, dcs)
  File "/home/itaru/x86/packages/nest-simulator/extras/help_generator/helpers.py", line 48, in check_ifdef
    require_reg = re.compile('HAVE\_((.*?)*)\n')
  File "/usr/lib64/python2.7/re.py", line 190, in compile
    return _compile(pattern, flags)
  File "/usr/lib64/python2.7/re.py", line 242, in _compile
    raise error, v # invalid expression
sre_constants.error: nothing to repeat
make[2]: *** [generate_help] Error 1
make[1]: *** [doc/CMakeFiles/generate_help.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by ikitayama** [comment]
Also the CMAKE options I used:
```cmake -DCMAKE_INSTALL_PREFIX:PATH=/home/itaru/x86/opt/nest-2.12.0 -Dwith-mpi=ON -Dwith-openmp=ON -Dwith-gsl=/home/itaru/x86/opt/gsl-2.3 -Dwith-python=OFF -Dwith-warning=OFF /home/itaru/x86/packages/nest-simulator
```

**Comment by ikitayama** [comment]
CMake command:

```
cmake -DCMAKE_INSTALL_PREFIX:PATH=/home/itaru/x86/opt/nest-2.12.0 -Dwith-mpi=ON -Dwith-openmp=ON -Dwith-gsl=/home/itaru/x86/opt/gsl-2.3 -Dwith-python=OFF -Dwith-warning=OFF /home/itaru/x86/packages/nest-simulator
```

**Comment by heplesser** [comment]
@ikitayama I think this is closely related to #712, which also describes a work-around. The proper fix is in #715. Briefly, the problem is as follows: The error you observe happens when then NEST online help is built. Since NEST 2.12, we use Python to build the help, even if NEST is compiled without Python. NEST the will typically use the system Python to build help, and if that is older than Python 2.7.8, it will fail because of a bug in the Python regular expression module. This is fixed in NEST 2.14.

If you need to build NEST 2.12, there are three options:
1. Install Python 2.7.8 or later and make sure that that Python is first in the `PATH` in the environment in which NEST is built.
1. Comment out parts of the CMakeLists.txt, as [described here](https://github.com/nest/nest-simulator/issues/712#issue-221544084)
1. Patch as in #715.

Please close the issue if this helps.

**Comment by ikitayama** [comment]
@heplesser Thanks! I took Option 1 and build wend through successfully.
